playground-common/setup-playground.sh

```diff
@@ -4,23 +4,30 @@
 # if we do structural changes in Playground's setup.
 
 function relativize() {
-    python -c "import os.path; print(os.path.relpath('$1', '$2'))"
+    python3 -c "import os.path; print(os.path.relpath('$1', '$2'))"
 }
 
 PLAYGROUND_REL_PATH=$(dirname $0)
 WORKING_DIR=$(pwd)
 
-# create gradle symlinks
-rm -rf gradle
-ln -s "${PLAYGROUND_REL_PATH}/gradle" gradle
-rm -rf gradlew
-ln -s "${PLAYGROUND_REL_PATH}/gradlew" gradlew
-rm -rf gradlew.bat
-ln -s "${PLAYGROUND_REL_PATH}/gradlew.bat" gradlew.bat
-rm -rf gradle.properties
-ln -s "${PLAYGROUND_REL_PATH}/androidx-shared.properties" gradle.properties
-rm -rf buildSrc
-ln -s "${PLAYGROUND_REL_PATH}/../buildSrc" buildSrc
+# helper symlink function that will also normalize the paths.
+function symlink() {
+    SRC=$1
+    TARGET=$2
+    TARGET_PARENT_DIR=$(dirname $TARGET)
+    REL_PATH_TO_TARGET_PARENT=$(relativize $SRC $WORKING_DIR/$TARGET_PARENT_DIR)
+    rm -rf $TARGET
+    ln -s $REL_PATH_TO_TARGET_PARENT $TARGET
```

Comment:
Just curious, for relative path, does `ln -sr` not work? I think it will automatically relativize the path based on original file.

Reply:
`ln: illegal option --r` so i guess it is not a thing on mac? will keep it as is.